src/oncall/scheduler/default.py

```diff
@@ -501,12 +448,20 @@ def populate(self, schedule, start_time, dbinfo):
         future_events = filter(lambda x: x != [], future_events)
         if future_events:
             first_event_start = min(future_events[0], key=lambda x: x['start'])['start']
-            cursor.execute('DELETE FROM event WHERE schedule_id = %s AND start >= %s', (schedule['id'], first_event_start))
+            query = 'DELETE FROM %s WHERE schedule_id = %s AND start >= %s' % (table_name, schedule['id'], first_event_start)
+            cursor.execute(query)
 
         # Create events in the db, associating a user to them
         for epoch in future_events:
-            user_id = self.find_next_user_id(schedule, epoch, cursor)
+            user_id = self.find_next_user_id(schedule, epoch, cursor, table_name)
             if not user_id:
                 continue
-            self.create_events(team_id, schedule['id'], user_id, epoch, role_id, cursor)
-        connection.commit()
\ No newline at end of file
+            self.create_events(team_id, schedule['id'], user_id, epoch, role_id, cursor, table_name)
+
+        # if this function is being called by preview return events
+        if req.method == 'GET':
+            resp.body = self.build_preview_response(req, resp, cursor, table_name)
+            query = "DROP TABLE %s;" % table_name
```

Comment:
We'd want to interpolate the table name the same way we do other queries, as arguments to `cursor.execute`

Reply:
There is no reason the table name needs to be interpolated here instead of just being hard coded so I changed it to a hard coded query.